### PR TITLE
Handle null value for String in Json when updating

### LINKED ISSUE #1344: update from JSON will skip null values
When `update` from JSON which contains property with null values, the property will be ignored.

To make the consistency with the `create` interface, corresponding field should be set to null.

### PR DESCRIPTION
**Pull request by beeender**
Fix #1344
Update the object's String field to empty string when the corresponding
field in Json is null.

### CODE REVIEW

**Comment by beeender** [comment]
@realm/java Review, thx!

**Comment by beeender** [comment]
retest this please

**Comment by beeender** [comment]
retest this please

File: realm-annotations-processor/src/main/java/io/realm/processor/RealmJsonTypeHelper.java [hunk 1]
```diff
@@ -163,22 +163,54 @@ private SimpleTypeConverter(String castType, String jsonType) {
...
-                .beginControlFlow("if (!json.isNull(\"%s\"))", fieldName)
-                    .emitStatement("obj.%s((%s) json.get%s(\"%s\"))",
-                        setter,
-                        castType,
-                        jsonType,
-                        fieldName)
-                .endControlFlow();
+            // TODO: This checking should be removed after nullable support for all types.
+            //       And the else block should be removed as well.
+            if (String.class.getName().equals(fieldType)) {
```
Comment (cmelchior) [inline realm-annotations-processor/src/main/java/io/realm/processor/RealmJsonTypeHelper.java]: Isn't there a `Utils.isString()` method?

File: realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java [hunk 2]
```diff
@@ -806,10 +806,20 @@ private void emitCreateUsingJsonStream(JavaWriter writer) throws IOException {
             String fieldName = field.getSimpleName().toString();
             String qualifiedFieldType = field.asType().toString();
 
-            if (i == 0) {
-                writer.beginControlFlow("if (name.equals(\"%s\") && reader.peek() != JsonToken.NULL)", fieldName);
+            if (String.class.getName().equals(qualifiedFieldType)) {
```
Comment (cmelchior) [inline realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java]: Used twice now, so if the Util method is missing. Would be good to re-add it.

**Comment by emanuelez** [comment]
LGTM :+1:

**Comment by cmelchior** [comment]
Apart from comments. Looks good :+1:

File: realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java (hunk 2 above)
Comment (cmelchior) [inline realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java]: Use helper method here as well :)

**Comment by cmelchior** [comment]
Also missing changelog update btw.

**Comment by cmelchior** [comment]
:+1:

**Comment by emanuelez** [comment]
:+1:

**Comment by beeender** [comment]
retest this please

**Comment by emanuelez** [comment]
retest this please

**Pull request merged**